common/payloads.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2020-2021 Studious Pancake
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "payloads.hpp"
+
+#include "reboot_to_payload.h"
+
+#include <cstring>
+
+namespace Payloads {
+
+    namespace {
+
+        bool LoadPayload(const char* path) {
```

Comment:
Maybe combine common functionality of this function with the one in `hekate.cpp`?

Reply:
done